pkg/common/commands/resetoffset/controller/offsets.go

```diff
@@ -129,27 +149,31 @@ func updateOffset(logger *zap.Logger,
 	offsetManager sarama.OffsetManager,
 	topic string,
 	partition int32,
-	offsetTime int64) (*kafkav1alpha1.OffsetMapping, error) {
+	offsetTime int64) (sarama.PartitionOffsetManager, *kafkav1alpha1.OffsetMapping, error) {
+
+	// Enhance The Logger With Partition
+	logger = logger.With(zap.Int32("Partition", partition))
+
+	// Get The New Offset Of Partition For Specified Time
+	newOffset, err := saramaClient.GetOffset(topic, partition, offsetTime)
+	if err != nil {
+		logger.Error("Failed to get Partition Offset for Time", zap.Int64("Time", offsetTime), zap.Error(err))
+		return nil, nil, err
+	}
 
 	// Create A PartitionOffsetManager For The Specified Partition
 	partitionOffsetManager, err := offsetManager.ManagePartition(topic, partition)
-	if err != nil {
+	if partitionOffsetManager == nil || err != nil {
 		logger.Error("Failed to create PartitionOffsetManager", zap.Error(err))
-		closePartitionOffsetManager(logger, partitionOffsetManager)
-		return nil, err
+		if partitionOffsetManager != nil {
+			partitionOffsetManager.AsyncClose()
+		}
```

Comment:
There's nothing technically wrong with this being here, but it seems odd for the caller to be responsible for closing an interface that was returned if there was an error also returned (which is the only way this AsynClose could be called).  I'm not sure I'd call for removing it as it doesn't hurt, but I am curious if there was there a path taken during testing that made this necessary?

Reply:
no, you're right it's not needed - will remove - thanks